src/app/webserver/webserver.component.ts

```diff
@@ -79,4 +89,40 @@ export class WebServerComponent {
     get service() {
         return this._service;
     }
+
+    get server(): Promise<WebServer> {
+        return new Promise<WebServer>((resolve, reject) => {
+            this._service.server.catch(e => {
+                if (e instanceof UnexpectedServerStatusError) {
+                    this._notifications.confirm(
+                        `Restart IIS Administration API`,
+                        `IIS Administration API is currently ${e.Status}, restart?`).then(confirmed => {
```

Comment:
Wording issue.

If the service is currently stopped, "restart" should be "start". :-)

Reply:
1. Fixed prompt message
2. If service started and fail to connect actually the user would be redirected to install page. We can talk about how we should better handle that case but case 2 would not happen for current behavior

Reply:
I have updated the question with the second sentence "Do you want to restart service"
I think the current way suffices, if the service is started and not being able to connect, the current message does show the information. I really don't know if we want to change the content of the message that much depending of the status